test/hotspot/jtreg/compiler/vectorapi/VectorRearrangeTest.java

```diff
@@ -0,0 +1,222 @@
+import java.util.Random;
+import jdk.test.lib.Utils;
+
+public class VectorRearrangeTest {
+    private static final int LENGTH = 2048;
+    private static final Random random = Utils.getRandomInstance();
+
+    private static final VectorSpecies<Byte> bspec128    = ByteVector.SPECIES_128;
+    private static final VectorSpecies<Short> sspec128   = ShortVector.SPECIES_128;
+    private static final VectorSpecies<Integer> ispec128 = IntVector.SPECIES_128;
+    private static final VectorSpecies<Long> lspec128    = LongVector.SPECIES_128;
+    private static final VectorSpecies<Float> fspec128   = FloatVector.SPECIES_128;
+    private static final VectorSpecies<Double> dspec128  = DoubleVector.SPECIES_128;
+    private static final VectorSpecies<Byte> bspec64     = ByteVector.SPECIES_64;
+    private static final VectorSpecies<Short> sspec64    = ShortVector.SPECIES_64;
+    private static final VectorSpecies<Integer> ispec64  = IntVector.SPECIES_64;
+    private static final VectorSpecies<Float> fspec64    = FloatVector.SPECIES_64;
+
+    private static byte[]   bsrc;
+    private static short[]  ssrc;
+    private static int[]    isrc;
+    private static long[]   lsrc;
+    private static float[]  fsrc;
+    private static double[] dsrc;
+
+    private static byte[]   bdst;
+    private static short[]  sdst;
+    private static int[]    idst;
+    private static long[]   ldst;
+    private static float[]  fdst;
+    private static double[] ddst;
+
+    private static int[][] indexes;
+
+    static {
+        bsrc = new byte[LENGTH];
+        ssrc = new short[LENGTH];
+        isrc = new int[LENGTH];
+        lsrc = new long[LENGTH];
+        fsrc = new float[LENGTH];
+        dsrc = new double[LENGTH];
+        bdst = new byte[LENGTH];
+        sdst = new short[LENGTH];
+        idst = new int[LENGTH];
+        ldst = new long[LENGTH];
+        fdst = new float[LENGTH];
+        ddst = new double[LENGTH];
+
+        for (int i = 0; i < LENGTH; ++i) {
+            bsrc[i] = (byte)random.nextInt();
+            ssrc[i] = (short)random.nextInt();
+            isrc[i] = random.nextInt();
+            lsrc[i] = random.nextLong();
+            fsrc[i] = random.nextFloat();
+            dsrc[i] = random.nextDouble();
+        }
+
+        int[] nums = {2, 4, 8, 16};
+        indexes = new int[4][];
+        for (int i = 0; i < 4; i++) {
+            indexes[i] = new int[nums[i]];
+            for (int j = 0; j < nums[i]; j++) {
+                indexes[i][j] = random.nextInt() & (nums[i] - 1);
+            }
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.REARRANGE_VB, IRNode.VECTOR_SIZE_8, " >0 "})
+    public void rearrange_byte64() {
+        VectorShuffle<Byte> shuffle = VectorShuffle.fromArray(bspec64, indexes[2], 0);
+        for (int i = 0; i < LENGTH; i += bspec64.length()) {
+            ByteVector.fromArray(bspec64, bsrc, i)
+                      .rearrange(shuffle)
+                      .intoArray(bdst, i);
+        }
+    }
```

Comment:
Is there any verification here that the result is correct? I usually do that with a `GOLD` value computed once at the beginning (in interpreter mode because C2 compilation only happens later), and then a `@Check` method where I compare the results.